shopfloor/services/location_content_transfer.py

```diff
@@ -253,9 +253,11 @@ def _find_location_to_work_from(self):
             [
                 ("picking_type_id", "in", self.picking_types.ids),
                 ("state", "=", "assigned"),
-                ("printed", "=", False),
+                "|",
+                ("user_id", "=", False),
+                ("user_id", "=", self.env.user.id),
             ],
-            order="create_date",
+            order="user_id, create_date",
```

Comment:
As you touch this line, let's fix it also. The right sorting should be:
```suggestion
            order="user_id, priority desc, scheduled_date asc, id desc",
```